linker-interface/shared/src/main/scala/org/scalajs/linker/interface/StandardConfig.scala

```diff
@@ -177,6 +186,38 @@ final class StandardConfig private (
   def withMaxConcurrentWrites(maxConcurrentWrites: Int): StandardConfig =
     copy(maxConcurrentWrites = maxConcurrentWrites)
 
+  /** Specifies whether to use the experimental WebAssembly backend.
+   *
+   *  When using this setting, the following settings must also be set:
+   *
+   *  - `withSemantics(sems)` such that the behaviors of `sems` are all set to
+   *    `CheckedBehavior.Unchecked`
+   *  - `withModuleKind(ModuleKind.ESModule)`
+   *  - `withOptimizer(false)`
+   *  - `withStrictFloats(true)` (this is the default)
+   *
+   *  These restrictions will be lifted in the future, except for the
+   *  `ModuleKind`.
+   *
+   *  If any of these restrictions are not met, linking will eventually throw
+   *  an `IllegalArgumentException`.
+   *
+   *  @note
+   *    The WebAssembly backend silently ignores `@JSExport` and `@JSExportAll`
+   *    annotations. All other language features are supported.
```

Comment:
I think at link time we should keep ignoring it. Otherwise we would have to hard-fail for the exports of `jl.Object.toString()` and `Throwable.{name,message}`. More generally it might be fine for some use cases to ignore them.

That said, I was thinking we could add a compiler option that warns about them. This way a codebase that actively uses `@JSExport` would be warned against it. I'm not yet sure about the best thing to do.

Reply:
OK. Let's defer that discussion for later. But it's maybe worth adding a note that this behavior might change in the future?